### PR TITLE
Fix cli update for Maven

### LINKED ISSUE #31629: Rename streamId to stream in user-facing update command
- `-DstreamId` -> `-Dstream`
- `--stream-id` -> `--stream`

as discussed with @aloubyansky and @ia3andy on Zulip.

### PR DESCRIPTION
**Pull request by ia3andy**
- `quarkus update` in a Maven project now delegates to the CLI version of the `quarkus-maven-plugin`
- `quarkus update` is not doing anything an shows a warning pointing to an issue for more info
- Fixes #31629 

```
~/D/q/code-with-quarkus> qss update -S 3.0
[INFO] Scanning for projects...
[INFO]
[INFO] ---------------------< org.acme:code-with-quarkus >---------------------
[INFO] Building code-with-quarkus 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- quarkus-maven-plugin:999-SNAPSHOT:update (default-cli) @ code-with-quarkus ---
[WARNING] quarkus:update goal is experimental, its options and output might change in future versions
[INFO] Looking for the newly published extensions in registry.quarkus.io
[INFO] Instructions to update this project from '2.16.4.Final' to '3.0.0.Alpha4':
[INFO]
[INFO] Recommended Quarkus platform BOM updates:
[INFO] Update: io.quarkus.platform:quarkus-bom:pom:2.16.4.Final -> 3.0.0.Alpha4
[INFO]
[INFO] Project update recipe has been created:
/var/folders/j8/sj85tl7d2pq_vnhbpj4t5lwm0000gn/T/quarkus-project-recipe-13722540005474724642.yaml
[INFO] The command to update this project:
mvn org.openrewrite.maven:rewrite-maven-plugin:4.39.0:run \
  -Drewrite.configLocation=/var/folders/j8/sj85tl7d2pq_vnhbpj4t5lwm0000gn/T/quarkus-project-recipe-13722540005474724642.yaml \
  -DactiveRecipes=io.quarkus.openrewrite.Quarkus -e
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.647 s
[INFO] Finished at: 2023-03-07T16:45:04+01:00
[INFO] ------------------------------------------------------------------------
```

### CODE REVIEW

**Review by gsmet — APPROVED** [review]
Looks good for the `--stream-id` -> `--stream` update, I will let the others validate the rest.

Thanks!

File: devtools/cli/src/main/java/io/quarkus/cli/build/GradleRunner.java [hunk 1]
```diff
@@ -136,22 +136,11 @@ public Integer projectInfo(boolean perModule) {
...
-            args.add("--stream-id");
-            args.add(targetPlatformStreamId);
-        }
-        if (perModule) {
-            args.add("--per-module");
-        }
-        return run(prependExecutable(args));
+        MessageWriter.info().error(
+                "quarkus update is not yet available for Gradle projects. See https://github.com/quarkusio/quarkus/issues/31658 for more information.");
+        return 1;
```
Comment (aloubyansky) [inline devtools/cli/src/main/java/io/quarkus/cli/build/GradleRunner.java:143]: We should still call the Gradle task to keep what we have, this is still useful. Or am I missing something?

File: devtools/cli/src/main/java/io/quarkus/cli/build/MavenRunner.java [hunk 2]
```diff
@@ -142,16 +147,21 @@ public Integer projectInfo(boolean perModule) throws Exception {
...
-    public Integer updateProject(String targetPlatformVersion, String targetPlatformStreamId, boolean perModule)
+    public Integer updateProject(String targetPlatformVersion, String targetPlatformStream, boolean perModule)
             throws Exception {
         ArrayDeque<String> args = new ArrayDeque<>();
         setMavenProperties(args, true);
-        args.add("quarkus:update");
+        final ExtensionCatalog extensionCatalog = ToolsUtils.resolvePlatformDescriptorDirectly(
+                ToolsConstants.QUARKUS_CORE_GROUP_ID, null,
+                Version.clientVersion(),
+                QuarkusProjectHelper.artifactResolver(), MessageWriter.info());
```
Comment (aloubyansky) [inline devtools/cli/src/main/java/io/quarkus/cli/build/MavenRunner.java:157]: We'll want to do that earlier and not based on the cli version. I can help with that in a follow up PR.

File: devtools/cli/src/main/java/io/quarkus/cli/build/GradleRunner.java (hunk 1 above)
Comment (aloubyansky) [reply devtools/cli/src/main/java/io/quarkus/cli/build/GradleRunner.java:143]: Doesn't the current impl still display instructions to update to the currently recommended versions? I am not talking about applying OR recipes.

File: devtools/cli/src/main/java/io/quarkus/cli/build/MavenRunner.java (hunk 2 above)
Comment (ia3andy) [reply devtools/cli/src/main/java/io/quarkus/cli/build/MavenRunner.java:157]: @aloubyansky  This is just for the quarkus maven plugin key resolution, we said we would use the CLI version for this right?

Comment (aloubyansky) [reply devtools/cli/src/main/java/io/quarkus/cli/build/MavenRunner.java:157]: The CLI version is not guaranteed to be recent enough. We'll need the target platform version.

**Pull request merged**